src/components/Form/Form.js

```diff
@@ -2,35 +2,55 @@ import React from 'react';
 
 import { Button } from '@components';
 
+import { Checkbox } from './Checkbox';
 import { Input, Textarea } from './Input';
+import { Select } from './Select';
 
 const NONE = 'NONE';
 const SUBMITTING = 'SUBMITTING';
 const SUCCESS = 'SUCCESS';
 const ERROR = 'ERROR';
 
 export const Form = ({
+  allowSubmit,
   error,
   fields,
   form,
+  onChange,
   onSubmit,
   submitting,
   success,
   ...props
 }) => {
   const [status, setStatus] = React.useState(NONE);
-  const [fieldState, setFieldState] = React.useState([]);
+  const [fieldState, setFieldState] = React.useState(
+    fields.reduce((acc, { defaultValue, name }) => {
+      acc[name] = typeof defaultValue !== 'undefined' ? defaultValue : '';
+      return acc;
+    }, {}),
+  );
 
-  const fieldData = fields.map((field, index) => ({
+  const fieldData = fields.map((field) => ({
     ...field,
     onChange: (e) => {
       // have to pull the value from the synthetic event here, because react throws it away
       const value = e.target.value;
-      setFieldState((prev) => Object.assign([], prev, { [index]: value }));
+      if (value !== fieldState[field.name]) {
+        setFieldState((prev) => ({
+          ...prev,
+          [field.name]: value,
+        }));
+      }
     },
-    value: fieldState[index] || field.defaultValue || '',
+    value: fieldState[field.name],
   }));
 
+  React.useEffect(() => {
+    if (onChange) {
+      onChange(fieldState);
```

Comment:
This is currently returning an object of `{ [field.name]: field.value }`, which is a) the format the form is using internally, and b) the format needed by the jobs page `onChange` callback; however, it is different to what gets passed to the `onSubmit` callback, which feels a little weird. The `onSubmit` gets an array of `[field.name, field.value]` tuples instead.

Could be worth updating the `onSubmit` I guess, or we could just leave it for now?